src/lib.rs

```diff
@@ -0,0 +1,230 @@
+    input: &'a str,
+    output: &'a str,
+    words: f64,
+    faces: f64,
+    actions: f64,
+    stutters: f64,
+    floating_rng: Xoshiro256Plus,
+    int_rng: Xoshiro256PlusPlus,
+    linkify: LinkFinder,
+}
+
+impl<'a> Default for UwUify<'a> {
+    fn default() -> Self {
+        Self {
+            text: "",
+            input: "",
+            output: "",
+            words: 1.0,
+            faces: 0.05,
+            actions: 0.125,
+            stutters: 0.225,
+            floating_rng: Xoshiro256Plus::seed_from_u64(69),
+            int_rng: Xoshiro256PlusPlus::seed_from_u64(420),
+            linkify: LinkFinder::new(),
+        }
+    }
+}
+
+impl<'a> UwUify<'a> {
+    pub fn new(
+        text: Option<&'a str>,
+        infile: Option<&'a str>,
+        outfile: Option<&'a str>,
+        words: Option<&'a str>,
+        faces: Option<&'a str>,
+        actions: Option<&'a str>,
+        stutters: Option<&'a str>,
+        random: bool,
+    ) -> UwUify<'a> {
+        // I dislike this
+
+        let mut linkify = LinkFinder::new();
+        linkify.kinds(&[LinkKind::Email, LinkKind::Url]);
+        linkify.url_must_have_scheme(false);
+        let mut uwuify = UwUify {
+            text: text.unwrap_or_default(),
+            input: infile.unwrap_or_default(),
+            output: outfile.unwrap_or_default(),
+            linkify,
+            ..Default::default()
+        };
+
+        if random {
+            uwuify.floating_rng = Xoshiro256Plus::seed_from_u64(rand::rngs::OsRng.next_u64());
+            uwuify.int_rng = Xoshiro256PlusPlus::seed_from_u64(rand::rngs::OsRng.next_u64());
+        }
+
+        if let Some(words) = words {
+            uwuify.words = words.parse::<f64>().unwrap();
+        }
+        if let Some(faces) = faces {
+            uwuify.faces = faces.parse::<f64>().unwrap();
+        }
+        if let Some(actions) = actions {
+            uwuify.actions = actions.parse::<f64>().unwrap();
+        }
+        if let Some(stutters) = stutters {
+            uwuify.stutters = stutters.parse::<f64>().unwrap();
+        }
+        uwuify
+    }
+
+    pub fn uwuify(&mut self) -> Result<(), Error> {
+        // Handle Text
+        if !self.text.is_empty() {
+            // Handle Text Output
+            if !self.output.is_empty() {
+                if Path::new(&self.output).exists() {
+                    return Err(Error::new(
+                        std::io::ErrorKind::AlreadyExists,
+                        format!("File '{}' already exists, aborting operation", &self.output),
+                    ));
+                }
+
+                let mut uwu_out_file = BufWriter::new(File::create(&self.output)?);
+                let uwu_progress_bar = progress_bar!();
+                self.uwuify_sentence(self.text, &mut uwu_out_file)?;
+
+                uwu_progress_bar.finish_with_message("UwU'ifying Complete!");
+                Ok(())
+            } else {
+                #[cfg(not(test))]
```

Comment:
What's the effect of this annotation?

Reply:
The compiler keeps/removes the line under that annotation, based on if I'm compiling a test build or not. The behavior I want is that when I'm compiling a test build, I want the program to write to a `std::io::sink()` (essentially the equivalent of `/dev/null`), and if I'm not compiling a test build, I want the program to write to `stdout`